compiler/codegen/ScratchRegisterManager.cpp

```diff
@@ -78,7 +78,6 @@ bool TR_ScratchRegisterManager::reclaimScratchRegister(TR::Register *reg)
 
 TR::Register *TR_ScratchRegisterManager::findOrCreateScratchRegister(TR_RegisterKinds rk)
    {
-   TR_ASSERT(rk != TR_VRF,"VRF: RA (findOrCreateScratchRegister) unimplemented");
```

Comment:
Main aim for Scratch Register is to provide a limited pool of scratch registers that can be used by evaluators without worrying about keeping track of them to add into register dependecy list or mark them dead. I did not seen in code that it needs to do anything special for vector registers. If it has requested a TR::Register of TR_VRF it needs to allocate register of that kind. Which should be handled here (https://github.com/eclipse/omr/blob/7694d089de6060a33d7b1a08fd092f7c40bf660e/compiler/codegen/ScratchRegisterManager.cpp#L105)
@fjeremic  I checked other utilities within scratch register manager and did not find any special attention required for vector registers. Do you know any other corner case because of which we have an assert here?

Reply:
I don't recall any restriction or why it is there. Does it just work™ if you enable it?

Reply:
Yes it works fine. 
I removed that  and used the scratch register manager to allocate vector register. Reason I needed to go through the scratch register manager is that there are so many parts for array copy and if we are generating tests for direction in codegen, we are left it with an Big ICF which includes both forward array copy and backward array copy. Scratch register Manager becomes handy in this case as generating dependencies for merge point we can use SRM utility function to add all the scratch registers allocated through it to dependency condition.